File: crowdfire/max_fuel.py
from math import ceil, floor

def maximum(speed, fuel, amount):
	maxi = -1
	for i in range(0, len(fuel)):
		temp = (amount * 1.0 / fuel[i]) * speed[i]
		maxi = max(temp, maxi)
	result = str(maxi)
	number, decimal = result.split(".")
	return number + "." + decimal[0:3]
	
n = int(raw_input())
speed = map(long, raw_input().split())
fuel  = map(long, raw_input().split())
amount = int(raw_input())

print maximum(speed, fuel, amount)

File: crowdfire/max_fuel.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
	int n;
	cin >> n;
	vector<double> speed(n);
	vector<double> fuel(n);
	for(int i=0; i<n; i++) cin >> speed[i];
	for(int i=0; i<n; i++) cin >> fuel[i];
	double amount;
	cin >> amount;
	double result = -1;
	for(int i=0; i<n; i++) {
		result = max(result, (amount * speed[i]) / fuel[i]);
	}
	result = (int)(result * 1000);
	result = (result * 1.0 / 1000);
	printf("%.3lf\n", result);
}